Automattic/wp-calypso
# Update Credit Card Logos

With #36850 and #36591 in mind, this PR refactors the credit card images we used and migrates them to Webpack.

## Changes proposed in this Pull Request

* Updated credit card image assets (both regular and disabled states) with detail, performance, and future color updates in mind.
    * Some images used older logos. Those were updated.
    * The markup in some of the older images was so convoluted, it made color updates unnecessarily tricky. Redrawing them simplified it.
* Updated `CreditCard`’s documentation to showcase all supported credit card types and fixed its selected state.
* Removed `CreditCardNumberInput`’s ability to display a credit card logo inside the field based on the input.
    * The supported credit card types are handled by Stripe in their `<iframe>` that overwrites our form fields. In result, we never used that feature anyway.
* Migrated all credit card-related images to Webpack.

### Before

<img src="https://user-images.githubusercontent.com/2060580/67113031-df1d9a80-f1d8-11e9-92be-a266a4dee9b7.png" width="484" height="596" alt="Before">

### After

<img src="https://user-images.githubusercontent.com/2060580/67113086-fc526900-f1d8-11e9-86c1-d6b887d38f1a.gif" width="484" height="596" alt="After">

## Testing instructions

These updates affect 3 components:

1. `CreditCard`
    * Visit the component’s documentation and make sure it looks like on the screenshot above. The _Add card_ button changes the selected item for easier testing.
1. `PaymentLogo`
    * Visit the component’s documentation and make sure the all credit card logos from the screenshot above are still listed (at their 50% size).
1. `CreditCardNumberInput`
    * In the dashboard, click on your avatar, and go to `Manage Purchases`.
    * Go to `Billing History` and `Add Credit Card`.
    * Enter one of the [test credit card details](https://stripe.com/docs/testing#cards) and perform a successful test transaction.

## Other comments

The remaining payment logos and the rest of images located in `static/images/upgrades` will be migrated to Webpack in a separate PR.
